packages/editor/src/components/document-outline/check.js

```diff
@@ -1,21 +1,19 @@
 /**
  * WordPress dependencies
  */
-import { withSelect } from '@wordpress/data';
+import { useSelect } from '@wordpress/data';
 import { store as blockEditorStore } from '@wordpress/block-editor';
 
-function DocumentOutlineCheck( { blocks, children } ) {
-	const headings = blocks.filter(
-		( block ) => block.name === 'core/heading'
-	);
+export default function DocumentOutlineCheck( { children } ) {
+	const hasHeadings = useSelect( ( select ) => {
+		const { getGlobalBlockCount } = select( blockEditorStore );
 
-	if ( headings.length < 1 ) {
+		return getGlobalBlockCount( 'core/heading' ) > 0;
+	} );
+
+	if ( hasHeadings ) {
```

Comment:
```suggestion
	if ( ! hasHeadings ) {
```

I think returning null when there are no headings is the expected behavior